src/main.rs

```diff
@@ -414,4 +415,20 @@ async fn start_component(
 		)
 		.await
 		.unwrap_or_else(|_| panic!("failed to start {}", cmd));
+	#[cfg(feature = "systemd")]
+	{
+		if *is_systemd_used() {
+			//currently pid is optional hence the double unwrap
+			let pids = process_manager.get_pid(key).await.unwrap().unwrap();
+			//spawn_scope takes a vec of pids in case we want to spawn a scope for multiple processes
+			spawn_scope(cmd.to_string(), vec![pids])
+				.await
+				.unwrap_or_else(|err| {
+					warn!(
+						"Failed to spawn scope for {}. Creating transient unit failed with {}",
+						cmd, err
+					);
+				});
```

Comment:
Imo, we should just log this error and continue.

Reply:
Oh wait, that is already what this is doing, I just saw the `unwrap` and got confused, but I guess `spawn_scope` returns a `Result<()>`.

Maybe we can replace the syntax here with something more obvious, like:
```rust
if let Err(err) = spawn_scope(cmd.to_string(), vec![pids]).await {
    warn!("Failed to spawn scope for {}. Creating transient unit failed with {}", cmd, err);
}
```